src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -37,6 +46,270 @@
+		char *constraintName = NameStr(constraintForm->conname);
+		constraintNameList = lappend(constraintNameList, pstrdup(constraintName));
+
+		heapTuple = systable_getnext(scanDescriptor);
+	}
+
+	systable_endscan(scanDescriptor);
+	table_close(pgConstraint, NoLock);
+
+	return constraintNameList;
+}
+
+
+/*
+ * WorkerFixPartitionConstraintCommandList creates a list of command strings that will fix
+ * all check constraint names of a shard.
+ */
+static List *
+WorkerFixPartitionConstraintCommandList(Oid relationId, uint64 shardId,
+										List *checkConstraintList)
+{
+	List *commandList = NIL;
+	Oid schemaId = get_rel_namespace(relationId);
+	char *schemaName = get_namespace_name(schemaId);
+	char *relationName = get_rel_name(relationId);
+	char *shardRelationName = pstrdup(relationName);
+
+	/* build shard relation name */
+	AppendShardIdToName(&shardRelationName, shardId);
+
+	char *quotedShardName = quote_qualified_identifier(schemaName, shardRelationName);
+
+	char *constraintName = NULL;
+	foreach_ptr(constraintName, checkConstraintList)
+	{
+		StringInfo shardQueryString = makeStringInfo();
+		appendStringInfo(shardQueryString,
+						 "SELECT worker_fix_partition_constraints(%s::regclass, "
+						 UINT64_FORMAT ", %s::text)",
+						 quote_literal_cstr(quotedShardName), shardId,
+						 quote_literal_cstr(constraintName));
+		commandList = lappend(commandList, shardQueryString->data);
+	}
+
+	return commandList;
+}
+
+
+/*
+ * RelationHasConstraint checks if a relation has a constraint with a given name.
+ */
+static bool
+RelationHasConstraint(Oid relationId, char *constraintName)
+{
+	bool found = false;
+
+	int scanKeyCount = 2;
+	ScanKeyData scanKey[2];
+
+	Relation pgConstraint = table_open(ConstraintRelationId, AccessShareLock);
+
+	ScanKeyInit(&scanKey[0], Anum_pg_constraint_conrelid,
+				BTEqualStrategyNumber, F_OIDEQ, ObjectIdGetDatum(relationId));
+	ScanKeyInit(&scanKey[1], Anum_pg_constraint_conname,
+				BTEqualStrategyNumber, F_NAMEEQ, CStringGetDatum(constraintName));
+
+	bool useIndex = false;
+	SysScanDesc scanDescriptor = systable_beginscan(pgConstraint, InvalidOid, useIndex,
+													NULL, scanKeyCount, scanKey);
+
+	HeapTuple heapTuple = systable_getnext(scanDescriptor);
+	if (HeapTupleIsValid(heapTuple))
+	{
+		found = true;
+	}
+
+	systable_endscan(scanDescriptor);
+	table_close(pgConstraint, NoLock);
+
+	return found;
+}
+
+
+/*
+ * RenameConstraintCommand creates the query string that will rename a constraint
+ */
+static char *
+RenameConstraintCommand(Oid relationId, char *constraintName, char *newConstraintName)
+{
+	char *qualifiedRelationName = generate_qualified_relation_name(relationId);
+	const char *quotedConstraintName = quote_identifier(constraintName);
+	const char *quotedNewConstraintName = quote_identifier(newConstraintName);
+
+	StringInfo renameCommand = makeStringInfo();
+	appendStringInfo(renameCommand, "ALTER TABLE %s RENAME CONSTRAINT %s TO %s;",
```

Comment:
minor: semicolon is redundant when executing a single command

Reply:
removed it, thanks for noticing